libs/domain/product/src/models/product.model.ts

```diff
@@ -31,6 +31,7 @@ export interface Product {
    * Ids of the category to which the product belongs.
    */
   categoryIds?: string[];
+  categories?: ProductCategory[];
```

Comment:
how much is this different from the `categoryIds`? Should we not only main category id's and resolve other data from a category service instead? Categories will play a larger role going forward (e.g. category list component, category driven navigations, etc.).

That being said, we can consider keeping them with products, as it's closely related.

Reply:
I would say, if we introduce a new service (ProductCategoriesService), to keep only `categoryIds` field has much more sense. What would this new service be? Will it just store categories from `concrete-product` and `catalog-search` requests and with ability to get them by ids? Need more details, because I learned about this feature for the first time.

One note, wee need to integrate this service with facets as well